sopt/src/main/scala/dagr/sopt/util/Configuration.scala

```diff
@@ -0,0 +1,248 @@
+      case t if t =:= typeOf[Short]        => config.getInt(path).toShort.asInstanceOf[T]
+      case t if t =:= typeOf[Int]          => config.getInt(path).asInstanceOf[T]
+      case t if t =:= typeOf[Long]         => config.getLong(path).asInstanceOf[T]
+      case t if t =:= typeOf[Float]        => config.getDouble(path).toFloat.asInstanceOf[T]
+      case t if t =:= typeOf[Double]       => config.getDouble(path).asInstanceOf[T]
+      case t if t =:= typeOf[BigInt]       => BigInt(config.getString(path)).asInstanceOf[T]
+      case t if t =:= typeOf[BigDecimal]   => BigDecimal(config.getString(path)).asInstanceOf[T]
+      case t if t =:= typeOf[Path]         => pathTo(config.getString(path)).asInstanceOf[T]
+      case t if t =:= typeOf[Duration]     => config.getDuration(path).asInstanceOf[T]
+      // TODO: replace this with better handling of List/Seq/Array
+      case t if t =:= typeOf[List[String]] => config.getStringList(path).toList.asInstanceOf[T]
+      case _ => throw new IllegalArgumentException("Don't know how to configure a " + typeOf[T])
+    }
+  }
+
+  /**
+    * Looks up a single value of a specific type in configuration. If the configuration key
+    * does not exist, an exception is thrown. If the requested type is not supported, an
+    * exception is thrown.
+    */
+  def configure[T : TypeTag](path: String) : T = {
+    Configuration.RequestedKeys += path
+
+    try {
+      asType[T](path)
+    }
+    catch {
+      case ex : Exception =>
+        error(s"#############################################################################")
+        error(s"Exception retrieving configuration key '$path': ${ex.getMessage}")
+        error(s"#############################################################################")
+        throw ex
+    }
+  }
+
+  /**
+    * Optionally accesses a configuration value. If the value is not present in the configuration
+    * a None will be returned, else a Some(T) of the appropriate type.
+    */
+  def optionallyConfigure[T : TypeTag](path: String) : Option[T] = {
+    Configuration.RequestedKeys += path
+    if (config.hasPath(path)) Some(configure[T](path))
+    else None
+  }
+
+  /**
+    * Looks up a value in the configuration, and if present returns it, otherwise returns the default
+    * value provided.
+    */
+  def configure[T : TypeTag](path: String, defaultValue: T) : T = {
+    Configuration.RequestedKeys += path
+    if (config.hasPath(path)) configure[T](path)
+    else defaultValue
+  }
+
+  /**
+    * Attempts to determine the path to an executable, first by looking it up in config,
+    * and if that fails, by attempting to locate it on the system path. If both fail then
+    * an exception is raised.
+    *
+    * @param path the configuration path to look up
+    * @param executable the default name of the executable
+    * @return An absolute path to the executable to use
+    */
+  def configureExecutable(path: String, executable: String) : Path = {
```

Comment:
I think perhaps this and configureExecutableFromBinDirectory belong in dagr's Configuration trait.